libfreerdp/core/codecs.c

```diff
@@ -100,14 +100,10 @@ BOOL freerdp_client_codecs_prepare(rdpCodecs* codecs, UINT32 flags)
 	return TRUE;
 }
 
-BOOL freerdp_client_codecs_reset(rdpCodecs* codecs, UINT32 flags,
-				 UINT32 width, UINT32 height)
+BOOL freerdp_client_codecs_reset(rdpCodecs* codecs, UINT32 flags, UINT32 width, UINT32 height)
 {
 	BOOL rc = TRUE;
 
-	if (!freerdp_client_codecs_prepare(codecs, flags))
```

Comment:
This might breaks h264 decoder. The idea is NOT to create codec contexts when creating the surface. Therefore we postpone the creation. Codec contexts are created when we receive a corresponding surface command. Therefore, only freerdp_client_codecs_prepare is called, freerdp_client_codecs_reset is never called. That at least breaks the h264 codec as weight and height is 0.

I think if we don't want to create context for unused codec, we should replace the FREERDP_CODEC_ALL flag in CreateSurface.

Now freerdp_client_codecs_reset in gdi_CreateSurface actually does nothing as no context has been created. 
freerdp_client_codecs_rese in gdi_ResetGraphics only have effect after a gfx->SurfaceCommand is received. But it seems we should have ResetGraphics command before any surface command received.

Does it still work with standard windows RDP server currently?